### ISSUE TITLE
[Mail Action] hash symbol in email password is interpreted as beginning of comment

### ISSUE THREAD

**Issue by macdroid53** [openhab/openhab1-addons#5645]
In the mail.cfg file.

If the password contains a # symbol, the password is truncated at the #.

Thus the password is invalid for access to the target email server.

**Comment by 9037568** [comment]
I created a special build of the action that drops a WARN into the log with the configured username and password. It shows the following when I configure the username `me` and password `me#you`:

````
20:58:06.477 [WARN ] [ction.mail.internal.MailActionService] - MailActionService configured with username = 'me', password = 'me#you'
````

This is with the 2.3.0 RC1. If the parsing of the config files has changed since then, it wasn't done here, AFAICS. 

Please provide details on which version you're running and on what platform. I then suggest raising this question in the community forum.

**Comment by macdroid53** [comment]
I put this here after discussing it on the forum just for future reference if a version is made for openhab 2.
I am on openhabian snapshot, so this is a OH1 binding in OH2. Sending to Gmail.

I tried wrapping the password with "" and '' and it did not send email.

If I removed the # (and the wrappers) the email was sent. With no other changes to the message, rule, or .cfg file.

Email to the same account and password with # works fine from a mail node  in node-red, an ssmtp on other PC's, and Android phones.
